pkg/edged/edged.go

```diff
@@ -327,6 +329,8 @@ func (e *edged) StartGarbageCollection() {
 	}, ImageGCPeriod, utilwait.NeverStop)
 
 	go utilwait.Until(func() {
+		e.podReadyLock.Lock()
```

Comment:
It is possible , but not much effective I feel since the only job this function would do is to return the e.podInitReady variable's value. Since the callers are only limited to 2 in this file , I feel it's ok for each function to acquire the lock , do the required stuff and release it . Abstracting locking / unlocking into a utility function seems somewhat less readable to me. Thoughts ?

Reply:
I can not see any readability issue if we abstract a function and do lock/unlock inside the function - callers will no need to worry about the data race stuff.

The function can help callers to avoid misusing the locks. For example, 

```
e.podReadyLock.RLock()
defer e.podReadyLock.RUnlock()
if e.initPodReady {	 		
  err := e.containerGCManager.GarbageCollect()
  ...
}
```

If we try to acquire the lock in `containerGCManager.GarbageCollect()` as well, there would be a deadlock problem. Right?

VS.

```
func (e*) IsInitPodReady() bool {
  e.lock()
  defer e.unlock
  return e.initPodReady
}

if e.IsInitPodReady() {
  err := e.containerGCManager.GarbageCollect()
}
```